### ISSUE TITLE
Memory leak on dx12 when drawing a simple triangle

### ISSUE THREAD

**Issue by Shaddaaa** [gfx-rs/gfx#3696]
Short info header:
- GFX version: 0.7.0
- OS: Windows
- GPU: Nvidia GeForce GTX 960

I tried to update the code of a [tutorial](https://github.com/Mistodon/gfx-hal-tutorials) from version 0.6 to 0.7, which lead to memory leaks (~4MB/s) that have not been there in version 0.6. After some testing I found out, that switching backend from dx12 to vulkan fixes the memory leaks, so there seems to be a problem with the dx12 backend.

Steps to reproduce:
```
1. git clone https://github.com/Shaddaaa/gfx_hal_test
2. cd gfx_hal_test
3. git checkout 8315268e286499445c604b19aacfb0019e5dfb0b
4. cargo run --bin part-1-triangle
```

**Comment by grovesNL** [comment]
Hi! :wave: This might be a duplicate of #3505, which was recently resolved by #3680 (on `master`)

**Comment by Shaddaaa** [comment]
Yep, can confirm it's fixed on master!
